framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1567,10 +1567,15 @@ protected void warnRedundantAnnotations(Tree tree, AnnotatedTypeMirror type) {
         }
 
         AnnotatedTypeMirror defaultAtms = atypeFactory.getDefaultAnnotations(tree, type);
+        if (defaultAtms.getAnnotations().isEmpty()) {
```

Comment:
What do you think of this alternative: move this up a bit in the method and check for `type.getKind() == Type.Kind.TYPEVAR` (exact syntax TBD, just the idea).
It's clearer to check for whether type is a type variable explicitly, instead of indirectly through the empty set of defaults. And we can make this check earlier in the method, before determining the defaults.

Reply:
Yes, that would make the code clearer. Thanks, I have already made the change.